### ISSUE TITLE
List block: add support for nested inner blocks

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#57272]
Pulling this out of a [recent blog post](https://geary.co/wordpress-block-editor-first-look/) and [video](https://www.youtube.com/watch?v=g8YxdSIq-uk) from a person using the Site Editor aiming for a specific layout as this would have saved some headaches.

**Comment by tellthemachines** [comment]
Thanks Anne! Did you happen to note the timestamp for the relevant part of the video? It's super long and I'm finding it quite confusing.

Edit: I'd mostly like a bit more context around the the reasoning for adding specifically flex and grid layout types to lists. The list block isn't really prepared to support much more than plain text lists (the most you can currently do is add an inline image instead of text, but that comes with its own accessibility issues). It's not even possible to remove the bullet/number styling right now. Is the goal to just render plain text list items in a row instead of vertically?

**Comment by annezazu** [comment]
I snagged this takeaway [from Riad's comment](https://geary.co/wordpress-block-editor-first-look/#comment-590). My understanding is that if the list block supported more layouts it would make creating what the video author did much easier to accomplish. I'd say the chunk from 29 min to 39 min is probably the best helpful for when the layout is being created.

**Comment by cbirdsong** [comment]
This would be a welcome addition, but I think it would not necessarily help with his problem? My read is he wanted to make something marked up as a list but with full inner block support. You can't make something like this and have it exist as an HTML list right now: 

![A list of upcoming events. Each item has the date in a colored block on the left and the time and name on multiple lines the right](https://github.com/WordPress/gutenberg/assets/1672206/6cc472b4-74b1-46a7-9cc9-be8d9ca713fb)

**Comment by tellthemachines** [comment]
Thanks Anne! @youknowriad, what did you have in mind here? As Cory and I mentioned above, List block is very limited in terms of content. Were you thinking in terms of allowing inner blocks inside List Items?

**Comment by youknowriad** [comment]
Yes, the screenshot above is what I had in mind, you can also have horizontal lists using flexbox and yes, this will only make sense if list item block (at least with flex/grid layout) support inner blocks other than paragraphs

**Comment by tellthemachines** [comment]
Updating the title because this is more about block nesting than adding layout support to List.
